openslides_backend/action/actions/meeting/import_.py

```diff
@@ -187,6 +188,37 @@ def replace_field_ids(
             entry[field] = [self.replace_map["user"][id_] for id_ in entry[field]]
         elif collection == "user" and field == "meeting_ids":
             entry[field] = list(self.replace_map["meeting"].values())
+        elif collection == "motion" and field == "recommendation_extension":
+            fqids_str = RECOMMENDATION_EXTENSION_REFERENCE_IDS_PATTERN.findall(
+                entry[field]
+            )
+            # split the entry in different parts.
+            entry_str = entry[field]
+            entry_list = []
+            for fqid in fqids_str:
+                search_str = "[" + fqid + "]"
+                idx = entry_str.find(search_str)
+                entry_list.append(entry_str[:idx])
+                entry_list.append(entry_str[idx : idx + len(search_str)])
+                entry_str = entry_str[idx + len(search_str) :]
+            entry_list.append(entry_str)
+            print("XXX", entry_list)
+            # replace the ids and generate the new_entry
+            new_entry = []
+            for tmp_entry in entry_list:
+                if tmp_entry.startswith("[") and tmp_entry.endswith("]"):
```

Comment:
I think this will lead to an error if the text uses `[]` without an fqid in it, e.g. just instead of normal brackets. Can't you just replace each id in the first loop above? Why do you even need this `entry_list`?

Reply:
I do now the replacement in the first loop, that is better.
I need to go thru the entry[field] string with the index, because if I only use replace, I could do errors with wrong replacement.
I think it is easier to split the string in a list, replace the fqids and then join it again.